### DISCUSSION TITLE
[STD] Fun mod idea: accept only one of the keys based on object position on playfield (KEY1 - Left side / KEY2 - Right side)

### DISCUSSION THREAD

**Discussion by cadr10** [ppy/osu#28389] [Ideas]
What I propose as a fun mod is having the keys assigned to a portion of the screen. KEY1 is assigned to the left side of the playfield, KEY2 the right side of the playfield. Using the wrong key results in a miss. Centered objects can be alternated between the two keys as they don't belong to neither sides, or belong to both sides, thus the alternating of the two keys.

How it would work:
 - KEY1 assigned to the left portion of the playfield.
 - KEY2 assigned to the right portion of the playfield.
 - To a better readability, circles that need to be clicked with KEY1 can be blue. Circles that need to be clicked with KEY2 can be red (colors are just example, but can be taken from skin or customized).
 - Player may customize its single tap ability in the customize mod menu. If two objects have a lower timing between them, the player will be able to use two keys to hit them (thinking of stream, mainly).
  - Centered objects will be alternated. If the last one was KEY1/Blue, the next will be KEY2/Red.
 
 It's simple, fun, the player will need to put the brain to works. I don't have any name idea, or at least good ones. (KS) Key side?

 Old maps would be fun to play. Newer maps coordination hell =)

**Comment by Octostomp** [comment]
What would happen along the middle line? If the map was made with grid snap then wouldn't there be a line directly in the middle? Would it be both keys, one, either?

**Reply by cadr10** [reply]
I thought it could be alternating style for the middle notes. If your last note was K2, the middle note will be K1, then the next middle note K2... or it can be the same key from the latest side, then alternate. K2 -> K2, then alternate. I think the alternating style is fitting because the middle notes belong to none of the sides while belong to both kind of feeling. 

Middle notes are open to discussion and what fits best to player readability and prediction, although I feel like they are kind of a wildcard atm like (you're single tapping both sides then have to alternate the middle ones).

**Reply by Octostomp** [reply]
Also, I think that you should be able to choose based on how many keys you use for osu!std. For example, if you have 3 keys, you can have it be split into 3 or just use the first 2 keys. Same with 4, except it would be more interesting since you could alternate within a side. Or is that impossible? IDK at this point.
